### PR TITLE
[WIP] feat: added client for redhat-sso

### PR DESCRIPTION
**Pull request by ziccardi**
## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has created for changes required on the client side

### CODE REVIEW

File: pkg/client/redhatsso/config.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	c.TokenEndpointURI = baseURL + "/auth/realms/" + c.Realm + "/protocol/openid-connect/token"
+	c.APIEndpointURI = baseURL + "/auth/realms/" + c.Realm
+}
+
+func NewRedhatSSOConfig() *RedhatSSOConfig {
+	cfg := &RedhatSSOConfig{
+		KafkaRealm: RealmConfig{
+			ClientIDFile:     "secrets/redhatsso-service.clientId",
+			ClientSecretFile: "secrets/redhatsso-service.clientSecret",
+		},
```
Comment (akoserwal) [inline pkg/client/redhatsso/config.go:33]: realm default value will be "redhat-external"

File: pkg/client/redhatsso/client.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+				{
+					URL: kc.config.KafkaRealm.APIEndpointURI,
+				},
+			},
+		}
+	}
+	return kc.configuration
+}
+
+func (kc *rhSSOClient) GetToken() (string, error) {
```
Comment (akoserwal) [inline pkg/client/redhatsso/client.go:66]: Can we  also add cache for token something like: GetCachedToken
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/keycloak/client.go#L178

File: pkg/client/redhatsso/client.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
+package redhatsso
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	serviceaccountsclient "github.com/redhat-developer/app-services-sdk-go/serviceaccounts/apiv1internal/client"
```
Comment (akoserwal) [inline pkg/client/redhatsso/client.go:7]: @ziccardi : please can you check the issue with import the sdk

**Comment by akoserwal** [comment]
Left some comments. great work @ziccardi

File: pkg/client/redhatsso/client.go (hunk 2 above)
Comment (ziccardi) [reply pkg/client/redhatsso/client.go:66]: done

Comment (ziccardi) [reply pkg/client/redhatsso/client.go:66]: done

File: pkg/client/redhatsso/config.go (hunk 1 above)
Comment (ziccardi) [reply pkg/client/redhatsso/config.go:33]: done

File: pkg/client/redhatsso/client.go (hunk 3 above)
Comment (ziccardi) [reply pkg/client/redhatsso/client.go:7]: @wtrocki any idea?

**Comment by wtrocki** [comment]
Working on this fix.

**Pull request closed without merging**